website/payments/admin.py

```diff
@@ -531,12 +531,147 @@ def export_csv(self, request: HttpRequest, queryset: QuerySet) -> HttpResponse:
     export_csv.short_description = _("Export")
 
 
+class BankAccountInline(admin.TabularInline):
+    model = BankAccount
+    fields = (
+        "iban",
+        "bic",
+        "mandate_no",
+        "valid_from",
+        "valid_until",
+        "last_used",
+    )
+    show_change_link = True
+
+    can_delete = False
+
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        return False
+
+
+class PaymentInline(admin.TabularInline):
+    model = Payment
+    fields = (
+        "created_at",
+        "type",
+        "amount",
+        "topic",
+        "notes",
+        "batch",
+    )
+
+    show_change_link = True
+
+    can_delete = False
+
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        return False
+
+
+class ThaliaPayEnabledFilter(admin.SimpleListFilter):
+    title = "Thalia Pay enabled"
+    parameter_name = "tpay_enabled"
+
+    def lookups(self, request, model_admin):
+        return ("Yes", "Yes"), ("No", "No")
+
+    def queryset(self, request, queryset):
+        value = self.value()
+        tpay_enabled = [x.id for x in PaymentUser.objects.all() if x.tpay_enabled]
+        if value == "Yes":
+            return PaymentUser.objects.filter(id__in=tpay_enabled)
+        if value == "No":
+            return PaymentUser.objects.exclude(id__in=tpay_enabled)
+        return None
+
+
+class ThaliaPayBalanceFilter(admin.SimpleListFilter):
+    title = "Thalia Pay balance"
+    parameter_name = "tpay_balance"
+
+    def lookups(self, request, model_admin):
+        return (
+            ("0", "€0,00"),
+            ("1", ">€0.00"),
+        )
+
+    def queryset(self, request, queryset):
+        value = self.value()
+        tpay_balance = [x.id for x in PaymentUser.objects.all() if x.tpay_balance != 0]
```

Comment:
You should be able to create 1 query here instead of 3:
1. You get the user
2. You calculate the balance
3. You get the users based on the balance

Or at least reduce it to 2.